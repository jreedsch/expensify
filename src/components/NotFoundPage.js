import React from 'react';
import { Link }  from 'react-router-dom';

const NotFoundPage = () => ( //implicit return
  // server side routing: <div> 404 = page not found! <a href="/">home</a></div>
  <div> 404 = page not found! <Link to='/'>Home</Link></div>
);

export default NotFoundPage;
