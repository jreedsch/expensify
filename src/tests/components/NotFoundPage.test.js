//import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import React from 'react';
//import toJSON from 'enzyme-to-json'; // see jest.config.json
import NotFoundPage from '../../components/NotFoundPage';

// in Jest, 'u' to save new baseline snapshot
test('should render NotFoundPage correctly', () => {
  const wrapper = shallow(<NotFoundPage />);
  //expect(toJSON(wrapper)).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
  //expect(wrapper.find('h1').length).toBe(1); //count of h1 elements
  //expect(wrapper.find('h1').text()).toBe('Expensify'); //contents of h1 element

  // using ReactShallowRenderer
  //const renderer = new ReactShallowRenderer();
  //renderer.render(<Header />);
  //console.log(renderer.getRenderOutput());
  //expect(renderer.getRenderOutput()).toMatchSnapshot();
})
