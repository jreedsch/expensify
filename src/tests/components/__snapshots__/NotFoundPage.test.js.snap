// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFoundPage correctly 1`] = `
<div>
   404 = page not found! 
  <Link
    replace={false}
    to="/"
  >
    Home
  </Link>
</div>
`;
